src/ua/ithillel/lesson4/test/java/MyLessonFourTest.java

```diff
@@ -0,0 +1,60 @@
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import part1.ArrayDataExeption;
+import part1.ArraySizeColumnException;
+import part1.ArraySizeRowException;
+
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrowsExactly;
+import static part1.ArrayValueCalculator.doCalc;
+
+
+public class MyLessonFourTest {
+    static String[][] firstArrayPositive = new String[][]{{"1", "2", "1", "2"}, {"1", "2", "1", "2"}, {"1", "2", "1",
+            "2"}, {"1", "2", "1", "2"}};
+    static String[][] secondArrayPositive = new String[][]{{"3", "7", "8", "1"}, {"5", "6", "9", "0"}, {"4", "4", "6",
+            "7"}, {"2", "2", "-3", "7"}};
+    static String[][] firstArrayWithDataException = new String[][]{{"3", "7", "a", "1"}, {"5", "6", "b", "0"}, {"4", "4",
+            "c", "7"}, {"2", "s", "-3", "7"}};
+    static String[][] secondArrayWithDataException = new String[][]{{"3", "7", "a", "1"}, {"5", "6", "b", "0"}, {"4", "4", "c",
+            "7"}, {"2", "s", "-3", "7"}};
+    static String[][] arrayWithMinRowExeption = new String[][]{{"3", "7", "8", "1"}, {"4", "4", "9", "7"}, {"2",
+            "9", "-3", "7"}};
+    static String[][] arrayWithMinColumnsExeption = new String[][]{{"3", "7", "8"}, {"4", "4", "9"}, {"2", "9", "-3"},
+            {"2", "9", "-3"}};
+    static String[][] arrayWithMaxRowExeption = new String[][]{{"3", "7", "8", "1"}, {"4", "4", "9", "7"}, {"2",
+            "9", "-3", "7"}, {"2", "9", "-3", "7"}, {"2", "9", "-3", "7"}};
+    static String[][] arrayWithMaxColumnsExeption = new String[][]{{"3", "7", "8", "7", "8"}, {"4", "4", "9", "7", "8"},
+            {"2", "9", "-3", "7", "8"}, {"2", "9", "-3", "7", "8"}};
+
+    @ParameterizedTest
+    @MethodSource("dataForCalcArrayNegative")
+    public void checkDoCalcNegative(String[][] inputArr, Exception ex) {
```

Comment:
```suggestion
    public void checkDoCalcNegative(String[][] inputArr, Class clazz) {
```
десь плюс-мінус так має бути можливість це робити